specification/containerregistry/resource-manager/Microsoft.ContainerRegistry/preview/2018-02-01-preview/containerregistry_build.json

```diff
@@ -0,0 +1,2242 @@
+      "required": [
+        "location"
+      ],
+      "properties": {
+        "id": {
+          "description": "The resource ID.",
+          "type": "string",
+          "readOnly": true
+        },
+        "name": {
+          "description": "The name of the resource.",
+          "type": "string",
+          "readOnly": true
+        },
+        "type": {
+          "description": "The type of the resource.",
+          "type": "string",
+          "readOnly": true
+        },
+        "location": {
+          "description": "The location of the resource. This cannot be changed after the resource is created.",
+          "type": "string",
+          "x-ms-mutability": [
+            "read",
+            "create"
+          ]
+        },
+        "tags": {
+          "description": "The tags of the resource.",
+          "type": "object",
+          "additionalProperties": {
+            "type": "string"
+          },
+          "x-ms-mutability": [
+            "read",
+            "create",
+            "update"
+          ]
+        }
+      },
+      "x-ms-azure-resource": true
+    },
+    "ProxyResource": {
+      "description": "The resource model definition for a ARM proxy resource. It will have everything other than required location and tags.",
+      "properties": {
+        "id": {
+          "description": "The resource ID.",
+          "type": "string",
+          "readOnly": true
+        },
+        "name": {
+          "description": "The name of the resource.",
+          "type": "string",
+          "readOnly": true
+        },
+        "type": {
+          "description": "The type of the resource.",
+          "type": "string",
+          "readOnly": true
+        }
+      },
+      "x-ms-azure-resource": true
+    },
+    "ErrorResponse": {
+      "description": "Azure container registry build API error object.",
+      "properties": {
+        "error": {
+          "$ref": "#/definitions/Error",
+          "description": "Azure container registry build API error body."
+        }
+      }
+    },
+    "Error": {
+      "description": "Azure container registry build API error body.",
+      "required": [
+        "code",
+        "message"
+      ],
+      "properties": {
+        "code": {
+          "description": "error code.",
+          "type": "string"
+        },
+        "message": {
+          "description": "error message.",
+          "type": "string"
+        }
+      }
+    },
+    "DockerBuildStep": {
+      "description": "The Docker build step.",
+      "type": "object",
+      "allOf": [
+        {
+          "$ref": "#/definitions/BuildStepProperties"
+        }
+      ],
+      "properties": {
+        "branch": {
+          "description": "The repository branch name.",
+          "type": "string"
+        },
+        "imageName": {
+          "description": "TODO (bindu): Backward-compatibility, bring back ImageName to support old CLI build extension, need to remove it.",
```

Comment:
Same.

Reply:
That was an interim extension and we should ask folks to update that once the official swagger is merged. SDK should not have this method. `imageName`